### PR TITLE
dApp: Unit tests for notifications store

### LINKED ISSUE #2149: Add unit tests to the store notifications module
## Description
Currently the notifications module of the store completely lacks tests. We need to add test coverage for that part of the code base.

## Acceptance criteria
- Cover the notification store code with unit tests.

## Tasks
- [ ]

### PR DESCRIPTION
**Pull request by taleldayekh**
Fixes #2149 

**Short description**
Adds missing tests for the notifications store.

**Definition of Done**

- [x] Steps to manually test the change have been documented
- [x] Acceptance criteria are met
- [ ] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1. Run the tests or simply trust the CI. :)

### CODE REVIEW

**Comment by taleldayekh** [comment]
Issue about removing the `NotificationImportance` property. #2173

File: raiden-dapp/tests/unit/store/notifications/getters.spec.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  test('can get all notifications sorted by newest first', () => {
+    const allNotifications = getters.notifications(notificationState);
+
+    expect(allNotifications[0]).toMatchObject({ id: 29 });
+    expect(allNotifications[1]).toMatchObject({ id: 22 });
+    expect(allNotifications[2]).toMatchObject({ id: 10 });
+  });
+
+  test('next notification id increments current highest id', () => {
+    const newNotificationId = getters.nextNotificationId(notificationState);
```
Comment (weilbith) [inline raiden-dapp/tests/unit/store/notifications/getters.spec.ts:25]: Could you please do an expect before this? So the test clearly shows that it was `29` before. And then do the "action" and very that this was actually causing the change. Thanks!

File: raiden-dapp/tests/unit/store/notifications/mutations.spec.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  ): NotificationsState => {
+    return {
+      notifications: { 1: notification },
+      newNotifications,
+    };
+  };
+
+  test('can delete notification', () => {
+    const state = createNotificationState();
+    const id = TestData.notifications.id;
```
Comment (weilbith) [inline raiden-dapp/tests/unit/store/notifications/mutations.spec.ts:22]: You stored it extra in a variable before. Why not using it now? :thinking:

File: raiden-dapp/tests/unit/store/notifications/mutations.spec.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    mutations.notificationDelete(state, id);
+    expect(state.notifications).toMatchObject({});
+  });
+
+  test('can set notifications to viewed', () => {
+    const state = createNotificationState(true);
+
```
Comment (weilbith) [inline raiden-dapp/tests/unit/store/notifications/mutations.spec.ts:30]: Please make sure it was `true` before and this test case does not "accidentally" succeed by previous test cases or similar.

**Review by weilbith — APPROVED** [review]
Thanks for adding those needed test. Cool typing you added there. :wink:
As I put at some places already, I would recommend to very the state of a test case before you do something. So we can be sure the test case works correct no matter any side effects of the test suit.
Else I think there is not much to say.

File: raiden-dapp/tests/unit/store/notifications/getters.spec.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      2: { ...notification, id: 10 },
+      3: { ...notification, id: 22 },
+    },
+    newNotifications: false,
+  };
+
+  test('can get all notifications sorted by newest first', () => {
+    const allNotifications = getters.notifications(notificationState);
+
+    expect(allNotifications[0]).toMatchObject({ id: 29 });
```
[the remark this answers is not in the document]
Comment (taleldayekh) [reply raiden-dapp/tests/unit/store/notifications/getters.spec.ts:19]: I like that approach, I'll make the change!

File: raiden-dapp/tests/unit/store/notifications/getters.spec.ts (hunk 1 above)
Comment (taleldayekh) [reply raiden-dapp/tests/unit/store/notifications/getters.spec.ts:25]: Done 👍

File: raiden-dapp/tests/unit/store/notifications/mutations.spec.ts (hunk 2 above)
Comment (taleldayekh) [reply raiden-dapp/tests/unit/store/notifications/mutations.spec.ts:22]: I overlooked that, fixed it now though.

File: raiden-dapp/tests/unit/store/notifications/mutations.spec.ts (hunk 3 above)
Comment (taleldayekh) [reply raiden-dapp/tests/unit/store/notifications/mutations.spec.ts:30]: Fixed it.

**Pull request merged**